src/Custom/Common/InternalListHelpers.cs

```diff
@@ -10,28 +11,40 @@ internal static class InternalListHelpers
     internal delegate Task<ClientResult> AsyncListResponseFunc(string continuationToken, int? pageSize);
     internal delegate ClientResult ListResponseFunc(string continuationToken, int? pageSize);
 
-    internal static AsyncPageableCollection<T> CreateAsyncPageable<T, U>(AsyncListResponseFunc listResponseFunc)
-        where U : IJsonModel<U>, IInternalListResponse<T>
+    internal static AsyncPageableCollection<T> CreateAsyncPageable<T,
+#if NET6_0_OR_GREATER
```

Comment:
I'll leave this to the discretion of the library owners, but usually a file like this one is added - https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Shared/TrimmingAttribute.cs so that these attributes don't need to be conditioned.

Reply:
Usually I use https://github.com/Sergio0694/PolySharp to avoid #ifdef, but in this case I think it would be overkill and really this one file is enough